ledger/participant-state/src/main/scala/com/daml/ledger/participant/state/v1/WriteParticipantPruningService.scala

```diff
@@ -35,10 +35,17 @@ trait WriteParticipantPruningService {
     * step upon successful completion of each prune call. To reach eventual consistency upon failures, be sure
     * to return ParticipantPruned even if the specified offset has already been pruned to allow ledger api server
     * index pruning to proceed in case of an earlier failure.
+    *
+    * @param pruneUpToInclusive The offset up to which contracts should be pruned.
+    * @param submissionId The submission id.
+    * @param pruneAllDivulgedContracts If set, instruct the ledger to prune all immediate and retroactive divulged contracts
+    *                                  created before `pruneUpToInclusive` independent of whether they were archived before
+    *                                  `pruneUpToInclusive`.
+    * @return The pruning result.
     */
   def prune(
       pruneUpToInclusive: Offset,
       submissionId: Ref.SubmissionId,
+      pruneAllDivulgedContracts: Boolean,
```

Comment:
Can you do this without modifying the v1 WriteService? We'd prefer to keep this stable, especially as it's going away soon.

Reply:
The `KeyValueParticipantStateWriter` still implements this interface, hence we need it for implementing full divulgence pruning on KV ledgers. I'll have a look, but not sure if I can circumvent this.

Reply:
How do you feel about waiting a couple of days until work is done to migrate `KeyValueParticipantStateWriter` to `v2.WriteService`?

Reply:
Unfortunately, this implementation needs to catch this week's snapshot release

Reply:
No worries, it happens.